### PR TITLE
Add systemd headers to our build system

### PR DESCRIPTION
**Pull request by chouquette**
### What does this PR do?

Add a recipe to provide systemd headers

### Motivation

We need those to build, and as such we should provide a recipe to provide those.
They currently are shipped in the build images, which is not the correct location for a source dependency. We need to be able to see those as a regular dependency and easily change the version we use.

### CODE REVIEW

**Comment by chouquette** [comment]
For a bit more context, this is extracted from https://github.com/DataDog/datadog-agent/pull/28262 as we don't have systemd headers in the new associated build images, and I can't find a compelling reason to add them there.
Doing it in the build system seems to be the better approach to me

**Comment by chouquette** [comment]
On a 2nd thought, maybe this shouldn't be split out. The image currently contains the systemd headers so it's not really possible to test this change appropriately :thinking:

File: omnibus/config/software/systemd.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+version("253") { source sha256: "acbd86d42ebc2b443722cb469ad215a140f504689c7a9133ecf91b235275a491" }
+
+ship_source_offer true
+
+source url: "https://github.com/systemd/systemd/archive/refs/tags/v#{version}.tar.gz"
+
+relative_path "#{name}-#{version}"
+
+build do
+  env = with_standard_compiler_flags(with_embedded_path)
```
Comment (alopezz) [inline omnibus/config/software/systemd.rb:35]: Unused, I think it's not needed.

```suggestion
```

File: omnibus/config/projects/iot-agent.rb [hunk 2]
```diff
@@ -65,6 +65,8 @@
   # creates required build directories
   dependency 'preparation'
 
+  dependency "systemd" if linux_target?
```
Comment (alopezz) [inline omnibus/config/projects/iot-agent.rb:68]: Should we perhaps move this to the datadog-iot-agent.rb software definition instead.

**Review by alopezz — COMMENTED** [review]
I'm fine with the change; my only concern is that we'll be adding include files to the package that we don't need (though we may already be doing that for other libraries too).

File: omnibus/config/software/systemd.rb (hunk 1 above)
Comment (chouquette) [reply omnibus/config/software/systemd.rb]: Indeed, that was a leftover of a time I tried to actually build systemd, before realizing it wasn't needed

File: omnibus/config/projects/iot-agent.rb (hunk 2 above)
Comment (chouquette) [reply omnibus/config/projects/iot-agent.rb]: That seems more sensible indeed, done!

**Review by alopezz — APPROVED** [review]
Fine with me; up to you if you keep this split or as part of the original PR.

**Comment by chouquette** [comment]
Let's keep it as part of the original PR, it's not that many change, and we can decide to resurrect this one if it makes the review easier

**Pull request closed without merging**